config/config_to_options.go

```diff
@@ -0,0 +1,298 @@
+package config
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/url"
+
+	broadcastclient "github.com/bitcoin-sv/go-broadcast-client/broadcast/broadcast-client"
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/engine/cluster"
+	"github.com/bitcoin-sv/spv-wallet/engine/datastore"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/taskmanager"
+	"github.com/bitcoin-sv/spv-wallet/engine/utils"
+	"github.com/bitcoin-sv/spv-wallet/metrics"
+	"github.com/go-redis/redis/v8"
+	"github.com/mrz1836/go-cachestore"
+	"github.com/rs/zerolog"
+)
+
+func (c *AppConfig) ToEngineOptions(logger zerolog.Logger, testMode bool) ([]engine.ClientOps, error) {
+	var options []engine.ClientOps
+
+	options = c.addUserAgentOpts(options)
+
+	options = c.addMetricsOpts(options)
+
+	options = c.addLoggerOpts(logger, options)
+
+	options = c.addDebugOpts(options)
+
+	options = c.addCacheStoreOpts(options)
+
+	var err error
+	if options, err = c.addClusterOpts(options); err != nil {
+		return nil, err
+	}
+
+	if options, err = c.addDataStoreOpts(options, testMode); err != nil {
+		return nil, err
+	}
+
+	options = c.addPaymailOpts(options)
+
+	options = c.addTaskManagerOpts(options)
+
+	options = c.addNotificationOpts(options)
+
+	options = c.addBroadcastClientOpts(options, logger)
```

Comment:
You forgot to apply `WithARC` which is the new way of ARC configuration for `ChainService`